python/rpdk/go/codegen.py

```diff
@@ -172,6 +182,23 @@ def generate(self, project):
             except (FileNotFoundError, CalledProcessError) as e:
                 raise DownstreamError("go fmt failed") from e
 
+        # Update settings as needed
+        need_to_write = False
+        for key, new in DEFAULT_SETTINGS.items():
+            old = project.settings.get(key)
+
+            if project.settings.get(key) != new:
+                LOG.debug(f"{key} version change from {old} to {new}")
+                project.settings[key] = new
+                need_to_write = True
+
+                if key == "pluginVersion":
+                    # Display any upgrade messages
+                    print(*check_version(old), sep="\n")
+
+        if need_to_write:
+            project._write_settings(LANGUAGE)
```

Comment:
I would much prefer we don't have to do this. I'll submit a PR to the main cloudformation-cli repo soon and change this if/when it goes in.

Reply:
https://github.com/aws-cloudformation/cloudformation-cli/pull/386